R/layer-sf.R

```diff
@@ -35,6 +35,18 @@ LayerSf <- ggproto("LayerSf", Layer,
         self$mapping$geometry <- as.name(geometry_col)
       }
     }
+
+    # automatically determine the legend type
+    if (is.na(self$show.legend) || isTRUE(self$show.legend)) {
+      if (is_sf(data)) {
+        if (sf_geometry_type(data) %in% c("POINT", "MULTIPOINT"))
+          self$geom_params$legend <- "point"
+        else if (sf_geometry_type(data) %in% c("LINESTRING", "MULTILINESTRING",
```

Comment:
You're calling `sf_geometry_type()` twice. Better to call it once, assign the result to a temp variable, and use that in the `if` statements.

Reply:
I am sorry that I can not express all my thanks in English! I feel that I am a young pupil and you are my teacher who teaches me hand by hand. Thanks!